### PR TITLE
Android style changes

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/leo/issues/579

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@fontsource-variable/inter": "5.0.16",
     "@fontsource/manrope": "5.0.18",
     "@fontsource/poppins": "5.0.8",
+    "@fontsource/roboto-flex": "^5.0.8",
```
Comment (fallaciousreasoning) [inline package.json:20]: nit: pin version
```suggestion
    "@fontsource/roboto-flex": "5.0.8",
```

File: package.json [hunk 2]
```diff
@@ -53,6 +54,7 @@
     "svelte-check": "3.6.6",
     "svelte-loader": "3.1.9",
     "svelte2tsx": "0.7.3",
+    "svg2vectordrawable": "^2.9.1",
```
Comment (fallaciousreasoning) [inline package.json:57]:
```suggestion
    "svg2vectordrawable": "2.9.1",
```

File: package.json [hunk 3]
```diff
@@ -78,6 +80,7 @@
     "skiafy-icons": "node ./src/scripts/gen-skia-icons.js",
     "check": "svelte-check --compiler-warnings a11y-click-events-have-key-events:ignore,a11y-no-static-element-interactions:ignore",
     "transform-tokens": "node ./src/tokens/transformTokens.js && tsc -p ./tsconfig-tokens.json",
+    "s2v": "mkdir -p ./tokens/android/drawable && npx s2v -f ./icons -o ./tokens/android/drawable",
```
Comment (fallaciousreasoning) [inline package.json:83]: Optional: Could be worth making a small `js` script, like `gen-skia-icons`, but completely optional

File: src/tokens/transformation/android/filters/lightColor.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const {
+  matchLightThemeToken
+} = require('../../common/tokenFilters')
+
+module.exports = (token) =>(matchLightThemeToken(token) || token.name.startsWith('color_primitive'))
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/filters/lightColor.js]: This doesn't seem quite right - all the primitive colors are included with the light colors?

File: src/tokens/transformation/android/colorName.js [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
   matcher: function (token) {
     return token.type === 'color'
   },
   transformer: function (token) {
-    return camelCase(token.path.slice(2).join(' '))
+    return token.name
+    .replace('color_light_','')
+    .replace('color_dark_','')
+    .replace('color_legacy_light_','')
+    .replace('color_legacy_dark_','')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/colorName.js]: Does this mean we don't strip the `color_` prefix from `color_primitive` colors?

File: src/tokens/transformation/android/filters/lightColor.js (hunk 4 above)
Comment (fallaciousreasoning) [reply src/tokens/transformation/android/filters/lightColor.js]: Could it be worth having three files:
1. Light
2. Dark
3. Primitive (which aren't themed)

File: src/tokens/transformation/android/formats/colorStyle.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      return (
+        `<color name="${compositeToken.name}">${colorCode}</color>`
+      )
+    })
+  return (
+    '<?xml version="1.0" encoding="utf-8"?>\n' +
+    fileHeader({ file, commentStyle: 'xml' }) +
+    '\n<resources>\n' +
+    fontStyles.join('\n') +
+    '\n</resources>\n'
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/colorStyle.js]: optional:

Might be nicer with string interpolation

```suggestion
    `
<?xml version="1.0" encoding="utf-8"?>
${fileHeader({ file, commentStyle: 'xml' })}
<resources>
  ${fontStyles.join('\n')}
</resources>`
```

File: src/tokens/transformation/android/formats/dimenStyle.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      return (
+        ` <dimen name="${dimenName}">${compositeToken.value}</dimen>`
+      )
+    })
+  return (
+    '<?xml version="1.0" encoding="utf-8"?>\n' +
+    fileHeader({ file, commentStyle: 'xml' }) +
+    '\n<resources>\n' +
+    fontStyles.join('\n') +
+    '\n</resources>\n'
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/dimenStyle.js]: optional nit: use string interpolation

File: src/tokens/transformation/android/formats/colorStyle.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+const { fileHeader } = require('style-dictionary').formatHelpers
+const { TinyColor } = require('@ctrl/tinycolor')
+
+module.exports = ({ dictionary, platform, options = {}, file }) => {
+  const fontStyles = dictionary.allTokens
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/colorStyle.js]: These are colors, not fontStyles

File: src/tokens/transformation/android/formats/dimenStyle.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+const { fileHeader } = require('style-dictionary').formatHelpers
+const changeCase = require('change-case')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/dimenStyle.js]: It doesn't look like we depend on this package. Ideally, we'd avoid depending on it at all

File: src/tokens/transformation/android/formats/fontStyle.js [hunk 10]
```diff
@@ -11,20 +11,14 @@ const printDescription = (description) =>
...
-    .filter(
-      (compositeToken) =>
-        compositeToken.original.value.textDecoration !== 'underline'
-    )
+    .filter((compositeToken) => compositeToken.type === 'custom-fontStyle')
     // create style
     .map((compositeToken) => {
       return (
-        `  <style name="${camelCase(compositeToken.name)}">\n` +
+        `  <style name="${changeCase.pascalCase(compositeToken.name.replace("font_android_", ""))}">\n` +
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/fontStyle.js]: Should this one also be `snake_case`?

Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/fontStyle.js]: Optional: use multiline string interpolation here so we can avoid all the concats/`\n`s

File: src/tokens/transformation/android/colorName.js [hunk 11]
```diff
@@ -1,11 +1,13 @@
...
   matcher: function (token) {
     return token.type === 'color'
   },
   transformer: function (token) {
-    return camelCase(token.path.slice(2).join(' '))
+    return token.name
+      .replace('color_light_', '')
+      .replace('color_dark_', '')
+      .replace('color_legacy_light_', '')
+      .replace('color_legacy_dark_', '')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/colorName.js:11]: I don't we're replacing `color_primitive` here

File: src/tokens/transformation/android/filters/lightColor.js [hunk 12]
```diff
@@ -0,0 +1,4 @@
+const { matchLightThemeToken } = require('../../common/tokenFilters')
+
+module.exports = (token) =>
+  matchLightThemeToken(token) || token.name.startsWith('color_primitive')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/filters/lightColor.js:4]: I think maybe we should have three sets of tokens
1. Light
2. Dark
3. Primitive

**Review by fallaciousreasoning — COMMENTED** [review]
Might need a `npm run format` but looking pretty good!

File: src/tokens/transformation/android/filters/lightColor.js (hunk 4 above)
Comment (deeppandya) [reply src/tokens/transformation/android/filters/lightColor.js]: @fallaciousreasoning in android, we can have one .xml for light colors and one for dark colors. and android would use light .xml as default colors so i have put primitive colors inside light .xml as those colors aren't themed.

File: src/tokens/transformation/android/colorName.js (hunk 5 above)
Comment (deeppandya) [reply src/tokens/transformation/android/colorName.js]: i have kept `color_` prefix only for primitive colors to differentiate in light .xml file.

File: src/tokens/transformation/android/formats/fontStyle.js (hunk 10 above)
Comment (deeppandya) [reply src/tokens/transformation/android/formats/fontStyle.js]: styles in chromium android are defined in pascalCase so  i kept it that way. 
for ref : https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/res/values/styles.xml;l=256?q=styles.xml&ss=chromium

File: src/tokens/transformation/android/colorName.js (hunk 5 above)
Comment (fallaciousreasoning) [reply src/tokens/transformation/android/colorName.js]: Could we maybe remove the `color_` - we'd still get the `primitive_` prefix on the color to differentiate, which is what we do for the CSS variables

File: src/tokens/transformation/android/filters/lightColor.js [hunk 13]
```diff
@@ -0,0 +1,4 @@
+const { matchLightThemeToken } = require('../../common/tokenFilters')
+
+module.exports = (token) =>
+  matchLightThemeToken(token) || token.name.startsWith('primitive')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/filters/lightColor.js:4]: Might be worth adding a comment about why we include the primitive colors here (for the people who aren't familiar with Android, like me :laughing:)

File: src/tokens/transformation/android/formats/fontStyle.js [hunk 14]
```diff
@@ -1,5 +1,5 @@
 const { fileHeader } = require('style-dictionary').formatHelpers
-const camelCase = require('../common/camelCaseHelper')
+const changeCase = require('change-case')
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/android/formats/fontStyle.js:2]: Hey, we still don't have a dependency on `change-case`. We should either:
1. Implement our own version of this (my preference)
2. Add the library to `package.json`

**Review by fallaciousreasoning — COMMENTED** [review]
Looking great! I think the `change-case` dependency is the last issue

File: src/tokens/transformation/android/formats/fontStyle.js (hunk 14 above)
Comment (deeppandya) [reply src/tokens/transformation/android/formats/fontStyle.js:2]: change-case  is included as a part of style-dictionary  : https://github.com/brave/leo/blob/4fd60db98b946355e38f781586502cd26d069b34/package-lock.json#L33326

(https://bravesoftware.slack.com/archives/D06HLN2SDJP/p1710861466483659)
https://github.com/brave/leo/blob/4fd60db98b946355e38f781586502cd26d069b34/package-lock.json#L8218

Comment (fallaciousreasoning) [reply src/tokens/transformation/android/formats/fontStyle.js:2]: cc'ing @petemill - should we depend on this directly? Not for this PR, just in general

**Review by fallaciousreasoning — APPROVED** [review]

File: src/scripts/gen-android-icons.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+  .then((icons) =>
+    icons.map(async (icon) => {
+      const svgContent = await fs.readFile(path.join(ICONS_FOLDER, icon), {
+        encoding: 'utf8'
+      })
+      svg2vectordrawable(svgContent).then((vectorContent) => {
+        const outputFileName = `ic_${path
+          .basename(icon, '.svg')
+          .replaceAll('-', '_')}.xml`
+        fs.writeFile(path.join(OUTPUT_FOLDER, outputFileName), vectorContent)
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply src/scripts/gen-android-icons.js:20]: This is fine - it's part of our build process and won't be exposed to user input

File: src/scripts/gen-android-icons.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+const ICONS_FOLDER = 'icons/'
+const OUTPUT_FOLDER = './tokens/android/drawable/'
+
+fs.mkdir(OUTPUT_FOLDER, { recursive: true })
+  .then(() => fs.readdir(ICONS_FOLDER))
+  .then((icons) => icons.filter((i) => i.endsWith('.svg')))
+  .then((icons) =>
+    icons.map(async (icon) => {
+      const svgContent = await fs.readFile(path.join(ICONS_FOLDER, icon), {
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply src/scripts/gen-android-icons.js:13]: This is fine.

**Comment by fallaciousreasoning** [comment]
@deeppandya not too important, but in future we try to squash and rebase when we merge :smile:

**Pull request merged**